lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2341,101 +2341,37 @@ func (d *zfs) readonlySnapshot(vol Volume) (string, revert.Hook, error) {
 		return "", nil, err
 	}
 
-	dataset := d.dataset(vol, false)
-	srcSnapshot := fmt.Sprintf("%s@temp_ro-%s", dataset, uuid.New())
+	snapshotOnlyName := fmt.Sprintf("temp_ro-%s", uuid.New())
```

Comment:
It feels like `readOnlySnapshot` should return the new temporary Volume that represents the read-only temporary snapshot, that would then allow using `UnmountVolumeSnapshot(vol)` on it, without the need for the new internal `unmountVolumeSnapshot()`. What do you think?

Reply:
Also, as the Volume has `SetTemporary` enabled on it, for clarity this function should be called something like `temporaryReadonlySnapshot()`.

Reply:
I've dropped the whole `SetTemporary` volume thing in favor of an additional argument in `mountVolumeSnapshot`.